simulate and test non terminal tasks

prat-gpt opened the pull request:
This PR allows integration tests from the monorepo to be run after the state changes are applied from a non terminal task to a local fork.

There was no clean way to apply the simulation to an anvil fork without doing a dry run first, then taking the state diff from the dry run, applying it to a network, and then running the integration tests.

blmalone commented on .circleci/config.yml:
```diff
@@ -228,6 +241,8 @@ workflows:
       - forge_build
       - forge_fmt
       - forge_test
+      #mororepo integration tests
```
```suggestion
      # monorepo integration tests
```

blmalone commented:
Can you please provide a PR description here to outline what feature is being built? Additionally, there don't seem to be any docs describing how to use this feature in the PR. Ideally I can read something and easily understand what's being built.

futurealisha committed:
Setup Rehearsal 2: Remove Signer and update the README.md (#477)

geoknee committed:
eth/028: executed (#483)

xenoliss committed:
sep/028: upgrade and re-initialize Base's SystemConfig to v2.3.0 (#476)

* upgrade Base's SystemConfig to v2.3.0

* comment simulate w/o ledger

* add explicit comment for gasLimit override

* mark as ready to sign

* push codeExceptions exceptions into HoloceneSystemConfigUpgrade contract (#478)

Some ready but not-yet-executed tasks depend on this contract, but they are simulated in CI so if that is green should be OK.

* add Tx 4 overview

* add context for extra tx

* remove mainnet task specific logic

* remove mainnet task specific logic

* remove .env comment

* impl nits feedback

* add cast td hints

* add storage layout link

* add more context for tx 4

* format VALIDATION.md and delete OVERVIEW.md

* fix validation formatting

* update validation to mention previous startBlock value

* update README.md

---------

Co-authored-by: George Knee <georgeknee@googlemail.com>
Co-authored-by: Matt Solomon <matt@mattsolomon.dev>

ElliotFriedman committed:
[PT: 2] FPS Import - Template Proposals (#479)

* remove: name and description from task config

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

add example task template config files

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

add fps example templates

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

remove name and description, set task nonce to current multisig nonce

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

fmt and add TODO to switch to manual nonce later

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

templates

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

remove safeNonce from mock network config

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* checkpoint

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* remove isNested flag from toml config file

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* fmt

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* fix is nested safe logic

* fix task01

* add Task02

* refactor _getDataToSign function

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add: gas template docs

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add: template examples to run

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add: error handle parsing l2 chain configs

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* fmt

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* remove unused variable name

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* error handle reading in and parsing task config file

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add breakdown of calldata difference for signers

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* remove setting scalars to simplify example task

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add duplicate call structure of task 020 in existing tooling

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* remove setting gas scalars from gas config template

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* note on commenting out failing check and link to diffchecker

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add: task config tests

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add forge build before running template

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* update docs to only include gas limits and not gas scalars

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* remove template ci runs

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* Update src/fps/FPS.md

Co-authored-by: blaine <blainemalone01@gmail.com>

* Update src/fps/FPS.md

Co-authored-by: blaine <blainemalone01@gmail.com>

* Update src/fps/FPS.md

* fps -> new task simulation tooling

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* remove unused template files

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* naming: fps.md -> readme.md

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* naming: move away from fps

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* use op types, remove inlined interfaces

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* update readme for running tasks

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* update: cleanup, make MultisigTask a generic template

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* fix test

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* delete: task config

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* update based on PR feedback, remove taskConfig.toml fields, and safeNonce and isNestedSafe

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* remove unused mock network configs

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* remove opmainnetConfig.toml

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add op set gas limit config

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* update command to use new run function

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* update command to only have mainnet config toml file

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* naming: networkConfigFilePath -> taskConfigFilePath

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* fmt

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* natspec, taskStorageWrites -> _taskStorageWrites

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* fix test

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add additional docs for address registry

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* fix: nested multisig issue

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add op gas config command

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* add task tests

* Revert "add task tests"

This reverts commit 70ed74f71a41956b276796dc36261782459b13f9.

* fix: nonce for nested safes use their own nonce and not parent

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* update natspec proposal -> task

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

---------

Signed-off-by: Elliot <elliotfriedman3@gmail.com>
Co-authored-by: prateek <prateekgupta105@gmail.com>
Co-authored-by: blaine <blainemalone01@gmail.com>

prat-gpt committed:
run non terminal tasks

ElliotFriedman committed:
Fix/naming (#486)

* naming: proposal -> task

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

* naming: proposal -> task

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

---------

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

ElliotFriedman committed:
update submodule (#490)

Signed-off-by: Elliot <elliotfriedman3@gmail.com>

prat-gpt committed:
add monorepo integraion tests to CI

prat-gpt committed:
remove justfile command

ElliotFriedman commented:
Closing as we already have merged this feature to the OP branch
